### PR TITLE
Add unit tests for WMI and WUA APIs

### PR DESCRIPTION
**Pull request by lchico**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/25767|

## Description

Unit tests were added to validate the WMI and WUA APIs integration for retrieving installed updates in Syscollector. To guarantee reliability, the tests ensure functionality under various scenarios, including error handling and edge cases. An abstract class was created to mock the WMI and WUA interactions, enabling comprehensive testing and simulation of different conditions.

- Implemented unit tests for QueryWMIHotFixes and QueryWUHotFixes functions.
- We covered different scenarios, including successful data retrieval, error handling, and edge cases.
- All unit tests pass successfully.

### Verified functionality on Windows systems

- [ ] Windows XP
- [x] Windows 10
- [x] Windows 11 
- [x] Windows Server 2003
- [x] Windows Server 2008 R2 
- [x] Windows Server 2012 R2 
- [x] Windows Server 2016
- [x] Windows Server 2019
- [x] Windows Server 2022 

## Coverage
Screenshot from 2024-11-22 00-19-32
image

### Verified functionality on Windows systems
- Windows 10
windows10_info

| **Wazuh Agent 4.8.1** | **Wazuh Agent 4.10-1 (Current PR)** |
|------------------------|------------------------------------|
|  |  |

**Note:** No errors in agent logs.

#### Windows Server 2019
win2019_version

| **Wazuh Agent 4.8.1** | **Wazuh Agent 4.10-1 (Current PR)** |
|------------------------|------------------------------------|
|  |  |

- Windows server 2016
version

**Note:** No errors in agent logs.

| **Wazuh Agent 4.8.1** | **Wazuh Agent 4.10-1 (Current PR)** |
|------------------------|------------------------------------|
|  |  |

**Note:** No errors in agent logs.

### CODE REVIEW

File: src/data_provider/src/sysInfoWin.cpp [hunk 1]
```diff
@@ -41,16 +41,10 @@
 #include "packages/packagesPYPI.hpp"
 #include "packages/packagesNPM.hpp"
 #include "packages/modernPackageDataRetriever.hpp"
+//#include "sysInfoWin.h"
```
Comment (lchico) [inline src/data_provider/src/sysInfoWin.cpp:44]: Remove.

File: src/data_provider/src/sysInfoWin.cpp [hunk 2]
```diff
@@ -942,64 +936,122 @@ void SysInfo::getPackages(std::function<void(nlohmann::json&)> callback) const
...
+    BSTR ConvertStringToBSTR(const char* str)
+    {
+        int len = MultiByteToWideChar(CP_ACP, 0, str, -1, NULL, 0);
+        BSTR bstr = SysAllocStringLen(0, len);
+        MultiByteToWideChar(CP_ACP, 0, str, -1, bstr, len);
+        return bstr;
+    }
+}
+
+std::string BstrToString(BSTR bstr)
```
Comment (lchico) [inline src/data_provider/src/sysInfoWin.cpp:994]: Add description.

File: src/data_provider/src/sysInfoWin.cpp [hunk 3]
```diff
@@ -942,64 +936,122 @@ void SysInfo::getPackages(std::function<void(nlohmann::json&)> callback) const
...
-static void QueryWMIHotFixes(std::set<std::string>& hotfixSet)
+void QueryWMIHotFixes(std::set<std::string>& hotfixSet, IComHelper& comHelper)
 {
     HRESULT hres;
     IWbemLocator* pLoc = NULL;
     IWbemServices* pSvc = NULL;
     std::ostringstream oss;
 
-    hres = CoCreateInstance(CLSID_WbemLocator, 0, CLSCTX_INPROC_SERVER, IID_IWbemLocator, (LPVOID*)&pLoc);
+    // hres = CoCreateInstance(CLSID_WbemLocator, 0, CLSCTX_INPROC_SERVER, IID_IWbemLocator, (LPVOID*)&pLoc);
```
Comment (lchico) [inline src/data_provider/src/sysInfoWin.cpp:1008]: Remove.

File: src/data_provider/src/sysInfoWin.cpp [hunk 4]
```diff
@@ -942,64 +936,122 @@ void SysInfo::getPackages(std::function<void(nlohmann::json&)> callback) const
...
 
     if (FAILED(hres))
     {
         oss << "WMI: Error creating IWbemLocator. Code: " << std::hex << hres;
         throw std::runtime_error(oss.str());
-        return;
     }
 
-    hres = pLoc->ConnectServer(_bstr_t(L"ROOT\\CIMV2"), NULL, NULL, 0, 0, 0, 0, &pSvc);
+    // hres = pLoc->ConnectServer(_bstr_t(L"ROOT\\CIMV2"), NULL, NULL, 0, 0, 0, 0, &pSvc);
```
Comment (lchico) [inline src/data_provider/src/sysInfoWin.cpp:1017]: Remove.

File: src/data_provider/src/sysInfoWin.cpp [hunk 5]
```diff
@@ -942,64 +936,122 @@ void SysInfo::getPackages(std::function<void(nlohmann::json&)> callback) const
...
+        if (pLoc) pLoc->Release();
+
         oss << "WMI: connection failed. Code: " << std::hex << hres;
         throw std::runtime_error(oss.str());
-        pLoc->Release();
-        return;
     }
 
-    hres = CoSetProxyBlanket(pSvc, RPC_C_AUTHN_WINNT, RPC_C_AUTHZ_NONE, NULL, RPC_C_AUTHN_LEVEL_CALL, RPC_C_IMP_LEVEL_IMPERSONATE, NULL, EOAC_NONE);
+    // hres = CoSetProxyBlanket(pSvc, RPC_C_AUTHN_WINNT, RPC_C_AUTHZ_NONE, NULL, RPC_C_AUTHN_LEVEL_CALL, RPC_C_IMP_LEVEL_IMPERSONATE, NULL, EOAC_NONE);
```
Comment (lchico) [inline src/data_provider/src/sysInfoWin.cpp:1028]: Remove.

**Comment by mjcr99** [comment]
## Test in WS2008r2
Environment:
* Agente 001: v.4.9.2
* Agente 002: 4.10.1 (Implemented fix)

sys_info 001:
```
sqlite> select * from sys_osinfo;
scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version|reference
0|2024/11/22 13:44:53|WS2008-1|x86_64|Microsoft Windows Server 2008 R2 Standard|6.1.7601||6|1||7601|windows||||sp1|1732283092874500000| |c717b80ea1dbb14b41f8aba9f5857eaf40ac19af
```

sys_info 002:
```
sqlite> select * from sys_osinfo;
scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version|reference
0|2024/11/22 13:55:48|WS2008|x86_64|Microsoft Windows Server 2008 R2 Standard|6.1.7601||6|1||7601|windows||||sp1|1732283742983875000| |c717b80ea1dbb14b41f8aba9f5857eaf40ac19af
```

sys_hotfixes 001:
```
sqlite> select * from sys_hotfixes;
scan_id|scan_time|hotfix|checksum
0|2024/11/22 13:44:55|KB2667402|e2a0ffe7ec1d90ff0c31ccd5d9e89af650815306
0|2024/11/22 13:44:55|KB2698365|de63f9ba1f1c246f8bb178caf07067511838a336
0|2024/11/22 13:44:55|KB2729094|f429b9885de20e0c822b56e06b5cb8193b526a24
0|2024/11/22 13:44:55|KB2813347|dd9d74778f1a688f46838c469755acb63f61a2ab
0|2024/11/22 13:44:55|KB2834140|c7ffaddc1b2923d84fe6f9967e70f8a85721113c
0|2024/11/22 13:44:55|KB2862330|00f3ab37e9eb0f758ce0ba1e367b11bea6435671
0|2024/11/22 13:44:55|KB2894844|c71db3e58cb20091119393cf18bc319c9ba7657d
0|2024/11/22 13:44:55|KB2900986|1264a46c8729588216e260e2a79038ad1aef5c0e
0|2024/11/22 13:44:55|KB2912390|c121acfbc21040e0559a14ff0b7e8359867ff38a
0|2024/11/22 13:44:56|KB2984972|fac96184ecad29c34e511e55586cb313c9cd399f
0|2024/11/22 13:44:56|KB3004375|2236f8ab3412bce48d3200014e0e1b24b9808bd2
0|2024/11/22 13:44:56|KB3018238|e8478be94e16ddfc80dc92e4360aca91ff87f7dc
0|2024/11/22 13:44:56|KB3020369|346f7fc6a446eb14b5f81217686ad06c2d796b23
0|2024/11/22 13:44:56|KB3035126|f59a74b7ea9594eb4f561596ab81855f664f5c23
0|2024/11/22 13:44:56|KB3035132|f2fdf52401f1036db4ecfd4ed3767734174fa0e5
0|2024/11/22 13:44:56|KB3046269|d5c2d7a057584b282813af7116b0187c033fd54d
0|2024/11/22 13:44:56|KB3059317|809e85da52455dbc4918bfa207786662e9aeffee
0|2024/11/22 13:44:56|KB3078601|09ae23caa2f12d03bb35a9d5f21008e345254adb
0|2024/11/22 13:44:57|KB3110329|b5f2697c5cd88a9a2fff4b76db77199b21c01cb5
0|2024/11/22 13:44:57|KB3124275|11cf47de2faa623a230443e970f2b44d2f2cd393
0|2024/11/22 13:44:57|KB3125574|d9b0e7e7dc34a04dbd1dbb17f9136ebf975f5d51
0|2024/11/22 13:44:57|KB3138612|ebd10c1c755b5e4442bfa784b49df722b768417f
0|2024/11/22 13:44:57|KB3156016|0d385e9cf3781dd1bfc390a250f9bf64bc818b5d
0|2024/11/22 13:44:57|KB3156019|cbc490af690c073de81cd255b6976a99d6e03eb9
0|2024/11/22 13:44:57|KB3159398|c5d8c0ca245152f39b2bdb403195a7da82971c7b
0|2024/11/22 13:44:57|KB3161949|8c20097ddd3c935d95773021630cd623d101416a
0|2024/11/22 13:44:57|KB3161958|6bff8f9a4d3a8beb56ce363889861df9a9f3cfbb
0|2024/11/22 13:44:58|KB3177467|2fb5206a88adad08c62849bd5d6c1f52d4a133fd
0|2024/11/22 13:44:58|KB4019990|f6c6545931131cbea07e0146232142a1a55f641c
0|2024/11/22 13:44:58|KB4040980|ddbf8c62e8e3ff2ac377d6d8497df4428251d16a
0|2024/11/22 13:44:58|KB4052978|62269c8420a893b95d277490b0d61a2752baec3a
0|2024/11/22 13:44:58|KB4054518|26abd87006124ce174d8a4c3364af1f905e51236
0|2024/11/22 13:44:58|KB976902|a3a2a50c3b0fcabba4d89ef2669d8ae21239840b
0|2024/11/22 13:44:58|KB976932|1c14fc86c845a15c8b6a08952afc30135b28a230
0|2024/11/22 13:44:58|KB976933|10b858a8d1ce886f91e5208921d483096de870d3
```

sys_hotfixes 002:
```
sqlite> select * from sys_hotfixes;
sqlite> 
```

There is no data retrieved using the implemented fix and the following log is found:
```
wazuh-modulesd:syscollector: ERROR: WUA: Error querying update history. Code 0x08240007
```

**Comment by jotacarma90** [comment]
# Windows server 2012 R
 - 4.10.1 (PR branch `enhancement/25767-write-unit-tests-for-wmi-api-integration`)
 - No errors agent logs: [logsW2012-4.10.1.txt](https://github.com/user-attachments/files/17872723/logsW2012-4.10.1.txt)

```
sqlite> select * from sys_osinfo;
0|2024/11/22 14:19:06|WINDOWS11|x86_64|Microsoft Windows 11 Enterprise Evaluation|10.0.22631.4169||10|0||22631.4169|windows||||2009|1732285146577499900|23H2|de514033f63e6c6a88d3b7d1005f74f65f40384b
```

```
sqlite> select * from sys_hotfixes;
0|2024/11/22 13:18:45|KB2267602|c363dfc47121b53dcef6c78aba83e0b507a62e6c
0|2024/11/22 13:18:45|KB2468871|8f99821b9e79bc2258cb56cd14cfcaf9bbeda8e5
0|2024/11/22 13:18:45|KB2478063|8511235ae3ab3b642d8ba429599092634fdde3a8
0|2024/11/22 13:18:45|KB2533523|b8c4cb9a2aeb6a64269e88f6116765420a65cd80
0|2024/11/22 13:18:46|KB2544514|8e468309f00c0f31b8a0f12a6a79d2658652e9e9
0|2024/11/22 13:18:46|KB2600211|cb13c01ba11045aabbb074fc3e61b0a3b2d88dd4
0|2024/11/22 13:18:46|KB2600217|8253af775746f8545784d27edb852281c9a06955
0|2024/11/22 13:18:46|KB4052623|43c4a7dd841771e4ac67b756a6301d6847e108f4
0|2024/11/22 13:18:46|KB5027397|6aa52f220a9aebb1004ed2c308b594e5b0b463d5
0|2024/11/22 13:18:46|KB5033204|2fb8a48c29e02391dbdef7ae5ee0140cf472e332
0|2024/11/22 13:18:46|KB5041655|ea4cb9e4ef3e79293a58a4f1bb421be97deda55b
0|2024/11/22 13:18:46|KB5042099|9aaedf5694deb2585b72c313643ecf80667aa69e
0|2024/11/22 13:18:46|KB5043076|56c7527a778bff40abc4f15ac3ba8373b6e266d1
0|2024/11/22 13:18:46|KB5043937|b754065c0ec8d3bb489e002358bb1488da44eeb3
```


- 4.9.2 (Released msi package)
 - No errors agent logs: [logsW2012-4.9.2.txt](https://github.com/user-attachments/files/17872725/logsW2012-4.9.2.txt)

```
sqlite> select * from sys_osinfo;
0|2024/11/22 15:07:33|WINDOWS11|x86_64|Microsoft Windows 11 Enterprise Evaluation|10.0.22631.4169||10|0||22631.4169|windows||||2009|1732288052946856700|23H2|de514033f63e6c6a88d3b7d1005f74f65f40384b
```

```
sqlite> select * from sys_hotfixes;
0|2024/11/22 15:07:33|KB2468871|8f99821b9e79bc2258cb56cd14cfcaf9bbeda8e5
0|2024/11/22 15:07:33|KB2478063|8511235ae3ab3b642d8ba429599092634fdde3a8
0|2024/11/22 15:07:34|KB2533523|b8c4cb9a2aeb6a64269e88f6116765420a65cd80
0|2024/11/22 15:07:34|KB2544514|8e468309f00c0f31b8a0f12a6a79d2658652e9e9
0|2024/11/22 15:07:34|KB2600211|cb13c01ba11045aabbb074fc3e61b0a3b2d88dd4
0|2024/11/22 15:07:34|KB2600217|8253af775746f8545784d27edb852281c9a06955
0|2024/11/22 15:07:34|KB5027397|6aa52f220a9aebb1004ed2c308b594e5b0b463d5
0|2024/11/22 15:07:34|KB5033204|2fb8a48c29e02391dbdef7ae5ee0140cf472e332
0|2024/11/22 15:07:34|KB5041655|ea4cb9e4ef3e79293a58a4f1bb421be97deda55b
0|2024/11/22 15:07:34|KB5042099|9aaedf5694deb2585b72c313643ecf80667aa69e
0|2024/11/22 15:07:34|KB5043076|56c7527a778bff40abc4f15ac3ba8373b6e266d1
```

**Comment by jotacarma90** [comment]
# Windows 11
 - 4.10.1 (PR branch `enhancement/25767-write-unit-tests-for-wmi-api-integration`)
 - No errors agent logs: [logsW11-4.10.1.txt](https://github.com/user-attachments/files/17872770/logsW11-4.10.1.txt)


```
sqlite> select * from sys_osinfo;
0|2024/11/22 14:19:06|WINDOWS11|x86_64|Microsoft Windows 11 Enterprise Evaluation|10.0.22631.4169||10|0||22631.4169|windows||||2009|1732285146577499900|23H2|de514033f63e6c6a88d3b7d1005f74f65f40384b
```

```
sqlite> select * from sys_hotfixes;
0|2024/11/22 13:18:45|KB2267602|c363dfc47121b53dcef6c78aba83e0b507a62e6c
0|2024/11/22 13:18:45|KB2468871|8f99821b9e79bc2258cb56cd14cfcaf9bbeda8e5
0|2024/11/22 13:18:45|KB2478063|8511235ae3ab3b642d8ba429599092634fdde3a8
0|2024/11/22 13:18:45|KB2533523|b8c4cb9a2aeb6a64269e88f6116765420a65cd80
0|2024/11/22 13:18:46|KB2544514|8e468309f00c0f31b8a0f12a6a79d2658652e9e9
0|2024/11/22 13:18:46|KB2600211|cb13c01ba11045aabbb074fc3e61b0a3b2d88dd4
0|2024/11/22 13:18:46|KB2600217|8253af775746f8545784d27edb852281c9a06955
0|2024/11/22 13:18:46|KB4052623|43c4a7dd841771e4ac67b756a6301d6847e108f4
0|2024/11/22 13:18:46|KB5027397|6aa52f220a9aebb1004ed2c308b594e5b0b463d5
0|2024/11/22 13:18:46|KB5033204|2fb8a48c29e02391dbdef7ae5ee0140cf472e332
0|2024/11/22 13:18:46|KB5041655|ea4cb9e4ef3e79293a58a4f1bb421be97deda55b
0|2024/11/22 13:18:46|KB5042099|9aaedf5694deb2585b72c313643ecf80667aa69e
0|2024/11/22 13:18:46|KB5043076|56c7527a778bff40abc4f15ac3ba8373b6e266d1
0|2024/11/22 13:18:46|KB5043937|b754065c0ec8d3bb489e002358bb1488da44eeb3
```


- 4.9.2 (Released msi package)
 - No errors agent logs: [logsW11-4.9.2.txt](https://github.com/user-attachments/files/17872774/logsW11-4.9.2.txt)


```
sqlite> select * from sys_osinfo;
0|2024/11/22 15:07:33|WINDOWS11|x86_64|Microsoft Windows 11 Enterprise Evaluation|10.0.22631.4169||10|0||22631.4169|windows||||2009|1732288052946856700|23H2|de514033f63e6c6a88d3b7d1005f74f65f40384b
```

```
sqlite> select * from sys_hotfixes;
0|2024/11/22 15:07:33|KB2468871|8f99821b9e79bc2258cb56cd14cfcaf9bbeda8e5
0|2024/11/22 15:07:33|KB2478063|8511235ae3ab3b642d8ba429599092634fdde3a8
0|2024/11/22 15:07:34|KB2533523|b8c4cb9a2aeb6a64269e88f6116765420a65cd80
0|2024/11/22 15:07:34|KB2544514|8e468309f00c0f31b8a0f12a6a79d2658652e9e9
0|2024/11/22 15:07:34|KB2600211|cb13c01ba11045aabbb074fc3e61b0a3b2d88dd4
0|2024/11/22 15:07:34|KB2600217|8253af775746f8545784d27edb852281c9a06955
0|2024/11/22 15:07:34|KB5027397|6aa52f220a9aebb1004ed2c308b594e5b0b463d5
0|2024/11/22 15:07:34|KB5033204|2fb8a48c29e02391dbdef7ae5ee0140cf472e332
0|2024/11/22 15:07:34|KB5041655|ea4cb9e4ef3e79293a58a4f1bb421be97deda55b
0|2024/11/22 15:07:34|KB5042099|9aaedf5694deb2585b72c313643ecf80667aa69e
0|2024/11/22 15:07:34|KB5043076|56c7527a778bff40abc4f15ac3ba8373b6e266d1
```

**Comment by lchico** [comment]
> ## Test in WS2008r2
> Environment:
> 
> * Agente 001: v.4.9.2
> * Agente 002: 4.10.1 (Implemented fix)
> 
> sys_info 001:
> 
> ```
> sqlite> select * from sys_osinfo;
> scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version|reference
> 0|2024/11/22 13:44:53|WS2008-1|x86_64|Microsoft Windows Server 2008 R2 Standard|6.1.7601||6|1||7601|windows||||sp1|1732283092874500000| |c717b80ea1dbb14b41f8aba9f5857eaf40ac19af
> ```
> 
> sys_info 002:
> 
> ```
> sqlite> select * from sys_osinfo;
> scan_id|scan_time|hostname|architecture|os_name|os_version|os_codename|os_major|os_minor|os_patch|os_build|os_platform|sysname|release|version|os_release|checksum|os_display_version|reference
> 0|2024/11/22 13:55:48|WS2008|x86_64|Microsoft Windows Server 2008 R2 Standard|6.1.7601||6|1||7601|windows||||sp1|1732283742983875000| |c717b80ea1dbb14b41f8aba9f5857eaf40ac19af
> ```
> 
> sys_hotfixes 001:
> 
> ```
> sqlite> select * from sys_hotfixes;
> scan_id|scan_time|hotfix|checksum
> 0|2024/11/22 13:44:55|KB2667402|e2a0ffe7ec1d90ff0c31ccd5d9e89af650815306
> 0|2024/11/22 13:44:55|KB2698365|de63f9ba1f1c246f8bb178caf07067511838a336
> 0|2024/11/22 13:44:55|KB2729094|f429b9885de20e0c822b56e06b5cb8193b526a24
> 0|2024/11/22 13:44:55|KB2813347|dd9d74778f1a688f46838c469755acb63f61a2ab
> 0|2024/11/22 13:44:55|KB2834140|c7ffaddc1b2923d84fe6f9967e70f8a85721113c
> 0|2024/11/22 13:44:55|KB2862330|00f3ab37e9eb0f758ce0ba1e367b11bea6435671
> 0|2024/11/22 13:44:55|KB2894844|c71db3e58cb20091119393cf18bc319c9ba7657d
> 0|2024/11/22 13:44:55|KB2900986|1264a46c8729588216e260e2a79038ad1aef5c0e
> 0|2024/11/22 13:44:55|KB2912390|c121acfbc21040e0559a14ff0b7e8359867ff38a
> 0|2024/11/22 13:44:56|KB2984972|fac96184ecad29c34e511e55586cb313c9cd399f
> 0|2024/11/22 13:44:56|KB3004375|2236f8ab3412bce48d3200014e0e1b24b9808bd2
> 0|2024/11/22 13:44:56|KB3018238|e8478be94e16ddfc80dc92e4360aca91ff87f7dc
> 0|2024/11/22 13:44:56|KB3020369|346f7fc6a446eb14b5f81217686ad06c2d796b23
> 0|2024/11/22 13:44:56|KB3035126|f59a74b7ea9594eb4f561596ab81855f664f5c23
> 0|2024/11/22 13:44:56|KB3035132|f2fdf52401f1036db4ecfd4ed3767734174fa0e5
> 0|2024/11/22 13:44:56|KB3046269|d5c2d7a057584b282813af7116b0187c033fd54d
> 0|2024/11/22 13:44:56|KB3059317|809e85da52455dbc4918bfa207786662e9aeffee
> 0|2024/11/22 13:44:56|KB3078601|09ae23caa2f12d03bb35a9d5f21008e345254adb
> 0|2024/11/22 13:44:57|KB3110329|b5f2697c5cd88a9a2fff4b76db77199b21c01cb5
> 0|2024/11/22 13:44:57|KB3124275|11cf47de2faa623a230443e970f2b44d2f2cd393
> 0|2024/11/22 13:44:57|KB3125574|d9b0e7e7dc34a04dbd1dbb17f9136ebf975f5d51
> 0|2024/11/22 13:44:57|KB3138612|ebd10c1c755b5e4442bfa784b49df722b768417f
> 0|2024/11/22 13:44:57|KB3156016|0d385e9cf3781dd1bfc390a250f9bf64bc818b5d
> 0|2024/11/22 13:44:57|KB3156019|cbc490af690c073de81cd255b6976a99d6e03eb9
> 0|2024/11/22 13:44:57|KB3159398|c5d8c0ca245152f39b2bdb403195a7da82971c7b
> 0|2024/11/22 13:44:57|KB3161949|8c20097ddd3c935d95773021630cd623d101416a
> 0|2024/11/22 13:44:57|KB3161958|6bff8f9a4d3a8beb56ce363889861df9a9f3cfbb
> 0|2024/11/22 13:44:58|KB3177467|2fb5206a88adad08c62849bd5d6c1f52d4a133fd
> 0|2024/11/22 13:44:58|KB4019990|f6c6545931131cbea07e0146232142a1a55f641c
> 0|2024/11/22 13:44:58|KB4040980|ddbf8c62e8e3ff2ac377d6d8497df4428251d16a
> 0|2024/11/22 13:44:58|KB4052978|62269c8420a893b95d277490b0d61a2752baec3a
> 0|2024/11/22 13:44:58|KB4054518|26abd87006124ce174d8a4c3364af1f905e51236
> 0|2024/11/22 13:44:58|KB976902|a3a2a50c3b0fcabba4d89ef2669d8ae21239840b
> 0|2024/11/22 13:44:58|KB976932|1c14fc86c845a15c8b6a08952afc30135b28a230
> 0|2024/11/22 13:44:58|KB976933|10b858a8d1ce886f91e5208921d483096de870d3
> ```
> 
> sys_hotfixes 002:
> 
> ```
> sqlite> select * from sys_hotfixes;
> sqlite> 
> ```
> 
> There is no data retrieved using the implemented fix and the following log is found:
> 
> ```
> wazuh-modulesd:syscollector: ERROR: WUA: Error querying update history. Code 0x08240007
> ```

Thanks, @mjcr99
I updated the code, and this issue should now be resolved. I also tested it on the same Windows Server 2008, and the logs no longer show the error. I was able to reproduce the log error, and with the new package, the issue is fixed. I will share my test results here.

Screenshot from 2024-11-22 16-59-03

| **Wazuh Agent 4.8.1**                             | **Wazuh Agent 4.10-1 (Current PR)**                  |
|---------------------------------------------------|-----------------------------------------------------|
|  |  |


**Note**: As expected, there are no new hotfixes, considering that this version of Windows may support these APIs.

**Comment by mjcr99** [comment]
Hello @lchico! As you said, after the fix, the hotfixes are obtained flawlessly:

```
root@server-ubu22:/home/vagrant# sqlite3 /var/ossec/queue/db/005.db
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
sqlite> select * from sys_hotfixes;
0|2024/11/25 13:23:54|KB2667402|e2a0ffe7ec1d90ff0c31ccd5d9e89af650815306
0|2024/11/25 13:23:54|KB2670838|c0942ef6032b6c5d6fe5e6fdff926774c8bc2909
0|2024/11/25 13:23:55|KB2698365|de63f9ba1f1c246f8bb178caf07067511838a336
0|2024/11/25 13:23:55|KB2729094|f429b9885de20e0c822b56e06b5cb8193b526a24
0|2024/11/25 13:23:55|KB2813347|dd9d74778f1a688f46838c469755acb63f61a2ab
0|2024/11/25 13:23:55|KB2834140|c7ffaddc1b2923d84fe6f9967e70f8a85721113c
0|2024/11/25 13:23:55|KB2841134|be29da1f14e1c7ecd0365cb97581ddb03f4124d7
0|2024/11/25 13:23:55|KB2849696|11dbd7c9398af055004630ac55884339d8583785
0|2024/11/25 13:23:55|KB2849697|ffe4d374150c775d3316e6207b487e51eac9b754
0|2024/11/25 13:23:55|KB2862330|00f3ab37e9eb0f758ce0ba1e367b11bea6435671
0|2024/11/25 13:23:55|KB2894844|c71db3e58cb20091119393cf18bc319c9ba7657d
0|2024/11/25 13:23:56|KB2900986|1264a46c8729588216e260e2a79038ad1aef5c0e
0|2024/11/25 13:23:56|KB2912390|c121acfbc21040e0559a14ff0b7e8359867ff38a
0|2024/11/25 13:23:56|KB2984972|fac96184ecad29c34e511e55586cb313c9cd399f
0|2024/11/25 13:23:56|KB3004375|2236f8ab3412bce48d3200014e0e1b24b9808bd2
0|2024/11/25 13:23:56|KB3018238|e8478be94e16ddfc80dc92e4360aca91ff87f7dc
0|2024/11/25 13:23:56|KB3020369|346f7fc6a446eb14b5f81217686ad06c2d796b23
0|2024/11/25 13:23:56|KB3035126|f59a74b7ea9594eb4f561596ab81855f664f5c23
0|2024/11/25 13:23:56|KB3035132|f2fdf52401f1036db4ecfd4ed3767734174fa0e5
0|2024/11/25 13:23:56|KB3046269|d5c2d7a057584b282813af7116b0187c033fd54d
0|2024/11/25 13:23:57|KB3059317|809e85da52455dbc4918bfa207786662e9aeffee
0|2024/11/25 13:23:57|KB3078601|09ae23caa2f12d03bb35a9d5f21008e345254adb
0|2024/11/25 13:23:57|KB3110329|b5f2697c5cd88a9a2fff4b76db77199b21c01cb5
0|2024/11/25 13:23:57|KB3124275|11cf47de2faa623a230443e970f2b44d2f2cd393
0|2024/11/25 13:23:57|KB3125574|d9b0e7e7dc34a04dbd1dbb17f9136ebf975f5d51
0|2024/11/25 13:23:57|KB3138612|ebd10c1c755b5e4442bfa784b49df722b768417f
0|2024/11/25 13:23:57|KB3156016|0d385e9cf3781dd1bfc390a250f9bf64bc818b5d
0|2024/11/25 13:23:57|KB3156019|cbc490af690c073de81cd255b6976a99d6e03eb9
0|2024/11/25 13:23:57|KB3159398|c5d8c0ca245152f39b2bdb403195a7da82971c7b
0|2024/11/25 13:23:57|KB3161949|8c20097ddd3c935d95773021630cd623d101416a
0|2024/11/25 13:23:58|KB3161958|6bff8f9a4d3a8beb56ce363889861df9a9f3cfbb
0|2024/11/25 13:23:58|KB3177467|2fb5206a88adad08c62849bd5d6c1f52d4a133fd
0|2024/11/25 13:23:58|KB4019990|f6c6545931131cbea07e0146232142a1a55f641c
0|2024/11/25 13:23:58|KB4040980|ddbf8c62e8e3ff2ac377d6d8497df4428251d16a
0|2024/11/25 13:23:58|KB4052978|62269c8420a893b95d277490b0d61a2752baec3a
0|2024/11/25 13:23:58|KB4054518|26abd87006124ce174d8a4c3364af1f905e51236
0|2024/11/25 13:23:58|KB976902|a3a2a50c3b0fcabba4d89ef2669d8ae21239840b
0|2024/11/25 13:23:58|KB976932|1c14fc86c845a15c8b6a08952afc30135b28a230
0|2024/11/25 13:23:58|KB976933|10b858a8d1ce886f91e5208921d483096de870d3
```

## Test in WS2022

Hotfixes are also obtained and no ERROR log are found under this OS:
```
root@server-ubu22:/home/vagrant# sqlite3 /var/ossec/queue/db/004.db
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
sqlite> select * from sys_hotfixes;
0|2024/11/25 13:23:56|KB2468871|8f99821b9e79bc2258cb56cd14cfcaf9bbeda8e5
0|2024/11/25 13:23:56|KB2478063|8511235ae3ab3b642d8ba429599092634fdde3a8
0|2024/11/25 13:23:56|KB2533523|b8c4cb9a2aeb6a64269e88f6116765420a65cd80
0|2024/11/25 13:23:56|KB2544514|8e468309f00c0f31b8a0f12a6a79d2658652e9e9
0|2024/11/25 13:23:56|KB2600211|cb13c01ba11045aabbb074fc3e61b0a3b2d88dd4
0|2024/11/25 13:23:57|KB2600217|8253af775746f8545784d27edb852281c9a06955
0|2024/11/25 13:23:57|KB5031993|36eecdba83b66d5ab93229e9e76bba895c97ce7f
0|2024/11/25 13:23:57|KB5032310|f1e86700590248a8cdcab96aa830e63d6b7ac10d
0|2024/11/25 13:23:57|KB5033118|9e4f8f02deafd9e6d76719a00179f808eedd92fc
```

**Review by MarcelKemp — APPROVED** [review]

**Comment by lchico** [comment]
# Windows Server 2003
Screenshot from 2024-11-25 10-12-05

| **Wazuh Agent 4.8.1** | **Wazuh Agent 4.10-1 (Current PR)** |
|------------------------|------------------------------------|
|  |  |

**Note**: We have detected new hotfixes that do not match the regular KB format.

- [Q147222](https://jeffpar.github.io/kbarchive/kb/147/Q147222/)
- [File 1](https://techcommunity.microsoft.com/blog/coreinfrastructureandsecurityblog/comparing-hotfixes-across-multiple-node-failover-clusters/255804#:~:text=Collecting%20Hotfixes%20and%20Node%20Names%2C%20placing%20them%20in%20a%20Array%20and%20Creating%20Unique%20Computer%20List) - This can be filtered but is not an error.

**Pull request merged**